### PR TITLE
Add foldMap to relevant data structure extension classes.

### PR DESCRIPTION
**Pull request by nuttycom**
(comment that was previously here no longer applies)

### CODE REVIEW

File: src/thx/Iterables.hx [hunk 1]
```diff
@@ -106,14 +106,14 @@ of its signature.
 /**
 Refer to `Array.map`.
 **/
-  inline public static function map<T, S>(it : Iterable<T>, f : T -> S) : Array<S>
-    return Iterators.map(it.iterator(), f);
+  inline public static function map<T, S>(it : Iterable<T>, f : T -> S) : Iterable<S>
```
Comment (fponticelli) [inline src/thx/Iterables.hx]: this and the following ones will break a lot of code. I am a little hesitant merging this.

Comment (fponticelli) [inline src/thx/Iterables.hx]: if we want to make this change we should also consider if there are other methods (there should be) that transform iterable/iterator into arrays and change them as well?

Comment (nuttycom) [inline src/thx/Iterables.hx]: Yeah, I'm not sure what the best thing is to do at this point, but I added this to the PR to point out the issue as much as anything. The general pattern of 'map' is that it returns a new data structure of the same type, and so I was surprised to see .map on an Iterator returning an array. The only code that might be broken is that which uses .map simply for its side-effects, and I think that's not a good practice in any case so hopefully it's pretty rare. That is why I added .forEach.

Iterator and Iterable are basically the only lazy collection types that Haxe has, so I think it's a bit of a shame to limit their utility by forcing their higher-order functions to be strict and evaluate the whole stream in order to transform it.

**Comment by nuttycom** [comment]
@fponticelli I have reworked this to just add foldMap and not modify the definition of 'map' in iterables.

**Pull request merged**